File: dart/2139.dart
import 'dart:io';

void main() {
  const List<int> ano = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 25];

  while (true) {
    var input = stdin.readLineSync();

    if (input == null) {
      break;
    }

    List<String> dm = input.split(" ");
    int i = 0;
    int x = 0;

    while (i < int.parse(dm[0]) - 1) {
      x += ano[i];
      i++;
    }

    x = 360 - x - int.parse(dm[1]);

    if (x > 1) {
      print("Faltam ${x} dias para o natal!");
    } else if (x == 1) {
      print("E vespera de natal!");
    } else if (x == 0) {
      print("E natal!");
    } else {
      print("Ja passou!");
    }
  }
}

File: python/2139.py

ano = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 25]

while True:
    try:
        mes, dia = map(int, input().split())
        x = 0
        i = 0
        
        while i < mes - 1:
            x += ano[i]
            i += 1
        x = 360 - x - dia
        if x > 1:
            print("Faltam %s dias para o natal!" %(x))
        elif x == 1:
            print("E vespera de natal!")
        elif x == 0:
            print("E natal!")
        elif x < 0:
            print("Ja passou!")
    except EOFError:
        break